include/jemalloc/internal/cache_bin.h

```diff
@@ -38,15 +38,8 @@ static const uintptr_t cache_bin_trailing_junk = JUNK_ADDR + 1;
  */
 extern const uintptr_t disabled_bin;
 
-/*
- * That implies the following value, for the maximum number of items in any
- * individual bin.  The cache bins track their bounds looking just at the low
- * bits of a pointer, compared against a cache_bin_sz_t.  So that's
- *   1 << (sizeof(cache_bin_sz_t) * 8)
- * bytes spread across pointer sized objects to get the maximum.
- */
-#define CACHE_BIN_NCACHED_MAX (((size_t)1 << sizeof(cache_bin_sz_t) * 8) \
-    / sizeof(void *) - 1)
+/* The maximum bytes in an individual bin should not exceed UINT16_MAX. */
+#define CACHE_BIN_NCACHED_MAX UINT16_MAX / sizeof(void *)
```

Comment:
cache_bin_sz_t is uint16. This looks the same, except the -1 (which we should keep)?

Reply:
yes, the value is actually almost the same. The integer division here will automatically downcast so I think we don't need the -1. Also no harm to add that of course.

Reply:
The original uses cache_bin_sz_t, which will correctly handle the difference if we ever change cache_bin_sz_t away from uint16_t. Why hard code it to UINT16_MAX here? I don't follow why this needs any change.

Reply:
Another tip: always use braces to cover the whole macro, otherwise it becomes unpredictable after the replacement (because of operator precedence for example).

Reply:
You are right. There is no necessity to make the change here. Changed it back.